### PR TITLE
Noah/SwiftUI Port

### PR DESCRIPTION
**Pull request by Npikielny**
# Overview
I'm porting my project from Cocoa to SwiftUI for both macOS and iOS. This first PR is just the bare bones–a view representable for both operating systems and a launch page.

# Changes Made
- Adds CatalogueItem (a view that gives a brief description and some images for a specific project)
- Adds ContentView for both iOS and macOS
- Adds RenderingView to interface between Renderers and the representable for the OS
- Adds Renderer boiler plate

# Testing
Manual coverage

# Screenshots

<details>

<summary>macOS UI</summary>

Screen Shot 2021-09-07 at 8 05 36 PM

</details>

<details>

<summary>iOS UI</summary>

Screen Shot 2021-09-07 at 8 05 20 PM

</details>

### CODE REVIEW

File: GraphicsHub/RendererCatalog/RendererCatalog.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                                                      // ("Tester Capped Renderer",TesterCappedRenderer.self),
+                                                      // ("Testing Inputs", TestInputRenderer.self)
+        ]
+        
+    
+    struct RenderingOption{
+        var description: String
+        var images: [Image]
+        var sources: [String]
+    }
```
Comment (Npikielny) [inline GraphicsHub/RendererCatalog/RendererCatalog.swift:34]: This is outdated

File: GraphicsHub/Renderer/3DRendering/CustomPathTracingRenderer/CustomPathRenderer.swift [hunk 2]
```diff
@@ -45,10 +45,6 @@ class CustomPathRenderer: HighFidelityRayRenderer {
     }
     
     override func draw(commandBuffer: MTLCommandBuffer, view: MTKView) {
-        let t = Float(frame) / 120 / 10
-        camera.rotation.y = -t * 360
-        camera.position.x = sin(t * 2 * Float.pi) * 15
-        camera.position.z = -cos(t * 2 * Float.pi) * 15
```
Comment (Npikielny) [inline GraphicsHub/Renderer/3DRendering/CustomPathTracingRenderer/CustomPathRenderer.swift:51]: This is outdated

File: GraphicsHub/RendererCatalog/RendererCatalog.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//
+//  Created by Noah Pikielny on 7/6/21.
+//
+
+import SwiftUI
+
+class RendererCatalog {
+    
+    static let catalog: [(String, RendererInfo.Type)] = [("Boid Simulation", BoidRenderer.self),
+    //                                                         ("Perlin Visualizer", PerlinNoiseRenderer.self),
```
Comment (spd75) [inline GraphicsHub/RendererCatalog/RendererCatalog.swift:13]: Why not put this commented line out at the bottom with the rest?

Comment (spd75) [inline GraphicsHub/RendererCatalog/RendererCatalog.swift:13]: Why not put this commented out line at the bottom with the rest?

File: SwiftUI/GraphicsHub/GraphicsHub/Catalogue/CatalogueItem.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                        }
+                    }
+                }.frame(height: 200)
+                .border(Color.primary, width: 3)
+                .cornerRadius(5)
+            }
+//            if let rendererType = renderer.renderer {
+//                NavigationLink("Start", destination: RendererDelegate(renderer: <#T##Renderer#>))
+//                FIXME: MTLDevice can't be created on startup...
+//            }
```
Comment (spd75) [inline SwiftUI/GraphicsHub/GraphicsHub/Catalogue/CatalogueItem.swift]: Are these commented lines needed?

File: SwiftUI/GraphicsHub/GraphicsHub/Catalogue/RendererCatalogue.swift [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+import SwiftUI
+
+class RendererCatalog {
+    
+    static let catalog: [RenderingOption] = [
+        RayTracingRenderer,
+        SlimeMoldRenderer,
+        ComplexRenderer,
+        // FIXME: Hide
```
Comment (spd75) [inline SwiftUI/GraphicsHub/GraphicsHub/Catalogue/RendererCatalogue.swift:16]: What is this fixme

File: SwiftUI/GraphicsHub/GraphicsHub/Renderer/SpecializedRenderers/Tests/TestRenderers.swift [hunk 6]
```diff
@@ -0,0 +1,16 @@
+//
+//  TestRenderers.swift
+//  GraphicsHub
+//
+//  Created by Noah Pikielny on 9/7/21.
+//
+
+import MetalKit
+
+class UVRenderer: Renderer {
```
Comment (spd75) [inline SwiftUI/GraphicsHub/GraphicsHub/Renderer/SpecializedRenderers/Tests/TestRenderers.swift:10]: All your other classes have spaces in between class header and body.  Keep format consistent

File: SwiftUI/GraphicsHub/GraphicsHub/GraphicsHubApp.swift [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+//  GraphicsHub
+//
+//  Created by Noah Pikielny on 9/7/21.
+//
+
+import SwiftUI
+
+@main
+struct GraphicsHubApp: App {
+    var body: some Scene {
```
Comment (spd75) [inline SwiftUI/GraphicsHub/GraphicsHub/GraphicsHubApp.swift:12]: Inconsistent spacing across project

File: SwiftUI/GraphicsHub/GraphicsHub/Catalogue/CatalogueItem.swift [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                        ForEach (0..<renderer.images.count) { index in
+                            let image = renderer.images[index]
+                            image
+                                .renderingMode(.original)
+                                .resizable()
+                                .frame(width: 155, height: 155)
+                                .cornerRadius(5)
+                        }
+                    }
+                }.frame(height: 200)
```
Comment (amyqxhuang) [inline SwiftUI/GraphicsHub/GraphicsHub/Catalogue/CatalogueItem.swift:35]: put on the next line? not sure what the convention is, but on line 23 you put it on the next line

File: SwiftUI/GraphicsHub/GraphicsHub/Catalogue/RendererCatalogue.swift [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        UVRendererOption
+    ]
+    
+    struct RenderingOption: Identifiable {
+        var id: String { name }
+
```
Comment (amyqxhuang) [inline SwiftUI/GraphicsHub/GraphicsHub/Catalogue/RendererCatalogue.swift:22]: delete space?

File: SwiftUI/GraphicsHub/GraphicsHub/Renderer/Renderer.swift [hunk 10]
```diff
@@ -0,0 +1,136 @@
+//
+//  Renderer.swift
+//  GraphicsHub
+//
+//  Created by Noah Pikielny on 9/7/21.
+//
+
+import MetalKit
+
+
```
Comment (amyqxhuang) [inline SwiftUI/GraphicsHub/GraphicsHub/Renderer/Renderer.swift:10]: delete space

**Review by amyqxhuang — APPROVED** [review]
looks good! 🤩🚀🤩🚀😎😎

File: GraphicsHub/RendererCatalog/RendererCatalog.swift [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+//  GraphicsHub
+//
+//  Created by Noah Pikielny on 7/6/21.
+//
+
+import SwiftUI
+
+class RendererCatalog {
+    
+    static let catalog: [(String, RendererInfo.Type)] = [("Boid Simulation", BoidRenderer.self),
```
Comment (danielVebman) [inline GraphicsHub/RendererCatalog/RendererCatalog.swift:12]: Not a fan of the spacing:

```
static let ... = [
  ...
]
```

File: GraphicsHub/RendererCatalog/RendererCatalog.swift [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+                                                             ("Ray Tracing", CustomRayTraceRenderer.self),
+                                                             ("Accelerated Ray Tracing", AcceleratedRayTraceRenderer.self),
+                                                      // ("Cornell Box", CornellBox.self),
+                                                      // ("Tester",TesterBaseRenderer.self),
+                                                      // ("Tester Capped Renderer",TesterCappedRenderer.self),
+                                                      // ("Testing Inputs", TestInputRenderer.self)
+        ]
+        
+    
+    struct RenderingOption{
```
Comment (danielVebman) [inline GraphicsHub/RendererCatalog/RendererCatalog.swift:30]:
```suggestion
    struct RenderingOption {
```

**Review by danielVebman — APPROVED** [review]
Looks good. There are a lot of styling inconsistencies but I didn't comment because for this it doesn't really matter.

**Pull request merged**